monix-reactive/shared/src/main/scala/monix/reactive/instances/CatsContravariantForObserver.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.instances
+import cats.Contravariant
+import monix.execution.Ack
+import monix.execution.Ack.Stop
+import monix.reactive.Observer
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+/** `cats.Contravariant` type class instance for [[monix.reactive.Observer Observer]].
+  *
+  * References:
+  *
+  *  - [[https://typelevel.org/cats/ typelevel/cats]]
+  */
+object CatsContravariantForObserver extends Contravariant[Observer] {
+  override def contramap[A, B](fa: Observer[A])(f: B => A): Observer[B] =
+    new ContravariantObserver(fa)(f)
+}
+
+private[reactive] class ContravariantObserver[A, B](source: Observer[A])(f: B => A) extends Observer[B] {
```

Comment:
Make this `final` and a `private` member of `object `CatsContravariantForObserver`.

`private[package]`  is useful if you plan on reusing it within `package`, but it's riskier than a plain `private`, because `private[package]` is a Scala construct, not a Java one.

Reply:
@alexandru Ok, corrected in in all three cases!